src/box/applier.cc

```diff
@@ -793,7 +797,10 @@ applier_parse_tx_row(struct applier_tx_row *tx_row)
 			diag_raise();
 		}
 	} else if (type == IPROTO_OK) {
-		/* Nothing to do. */
+		if (xrow_decode_heartbeat(row, NULL,
+					  &tx_row->req.vclock_sync) != 0) {
+			diag_raise();
```

Comment:
Nit: You could use `xrow_decode_heartbeat_xc` here.

Reply:
Done. See the diff above.